walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryExtraTest.java

```diff
@@ -0,0 +1,392 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** */
+@RunWith(JUnit4.class)
+public final class FindMeetingQueryExtraTest {
+  private static final Collection<Event> NO_EVENTS = Collections.emptySet();
+  private static final Collection<String> NO_ATTENDEES = Collections.emptySet();
+
+  // Some people that we can use in our tests.
+  private static final String PERSON_A = "Person A";
+  private static final String PERSON_B = "Person B";
+  private static final String PERSON_C = "Person C";
+  private static final String PERSON_D = "Person D";
+  private static final String PERSON_E = "Person E";
+  private static final String PERSON_F = "Person F";
+
+  // All dates are the first day of the year 2020.
+  private static final int TIME_0800AM = TimeRange.getTimeInMinutes(8, 00);
+  private static final int TIME_0830AM = TimeRange.getTimeInMinutes(8, 30);
+  private static final int TIME_0845AM = TimeRange.getTimeInMinutes(8, 45);
+  private static final int TIME_0900AM = TimeRange.getTimeInMinutes(9, 00);
+  private static final int TIME_0930AM = TimeRange.getTimeInMinutes(9, 30);
+  private static final int TIME_1000AM = TimeRange.getTimeInMinutes(10, 00);
+  private static final int TIME_1015AM = TimeRange.getTimeInMinutes(10, 15);
+  private static final int TIME_1030AM = TimeRange.getTimeInMinutes(10, 30);
+  private static final int TIME_1100AM = TimeRange.getTimeInMinutes(11, 00);
+  private static final int TIME_1130AM = TimeRange.getTimeInMinutes(11, 30);
+  private static final int TIME_1200PM = TimeRange.getTimeInMinutes(12, 00);
+  private static final int TIME_0330PM = TimeRange.getTimeInMinutes(15, 30);
+  private static final int TIME_0600PM = TimeRange.getTimeInMinutes(18, 00);
+
+  private static final int DURATION_15_MINUTES = 15;
+  private static final int DURATION_30_MINUTES = 30;
+  private static final int DURATION_60_MINUTES = 60;
+  private static final int DURATION_90_MINUTES = 90;
+  private static final int DURATION_1_HOUR = 60;
+  private static final int DURATION_2_HOUR = 120;
+
+  private FindMeetingQuery query;
+
+  @Before
+  public void setUp() {
+    query = new FindMeetingQuery();
+  }
+
+  @Test
+  public void singleAttendeeOverlappingEvents() {
+    // Have an overlapping event for a person. We should only see two options.
+    //
+    // Events  :       |--A--|
+    //                     |--A--|
```

Comment:
I'm wondering if this case will happen in reality. Usually a person can only attend one meeting at a specific time point. But it should be fine to consider this, e.g. what if the request has some errors. If your service can handle this, it means it is more fault-tolerant.